pkg/api/http/client.go

```diff
@@ -312,6 +334,26 @@ func (c Client) doJSON(ctx context.Context, method string, url string, req []byt
 }
 
 func (c Client) do(ctx context.Context, method string, url string, req []byte, opts ReqOpts) ([]byte, http.Header, error) {
+	// Compress the body if opts.CompressBody is true, the request is big enough,
+	// and there isn't already a Content-Encoding set on the request (meaning that
+	// it was already compressed upstream).
+	if opts.CompressBody &&
+		len(req) >= compressionSizeThreshold &&
+		opts.Headers["Content-Encoding"] == "" {
```

Comment:
IMO, let's do this + remove the TODO?

```suggestion
	contentEncoding = c.Headers["Content-Encoding"]
	if c := opts.Headers["Content-Encoding"]; c != "" {
		contentEncoding = c
	}
	// Compress the body if opts.CompressBody is true, the request is big enough,
	// and there isn't already a Content-Encoding set on the request (meaning that
	// it was already compressed upstream).
	if opts.CompressBody &&
		len(req) >= compressionSizeThreshold &&
		contentEncoding == "" {
```

Reply:
Done.